server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorFactory.java

```diff
@@ -111,7 +111,10 @@ private static TermsAggregatorSupplier bytesSupplier() {
                 execution = ExecutionMode.fromString(executionHint);
             }
             // In some cases, using ordinals is just not supported: override it
-            if (valuesSource.hasOrdinals() == false || matchNoDocs(context, parent)) {
+            if (valuesSource.hasOrdinals() == false) {
+                execution = ExecutionMode.MAP;
+            }
+            if (matchNoDocs(context, parent) && bucketCountThresholds.getMinDocCount() > 0) {
```

Comment:
I wonder if this should not kick in if the user sends a hint.  We can run this in Global Ordinals mode, even if it's expensive, and if the user is asking to do that, it might be for a good reason.

Reply:
I pushed a patch.